app/src/features/apiClient/screens/apiClient/components/clientView/APIClientView.tsx

```diff
@@ -256,6 +275,24 @@ const APIClientView: React.FC<Props> = ({ apiEntry, notifyApiRequestFinished })
     trackRQDesktopLastActivity(API_CLIENT.REQUEST_SENT);
   }, [entry, appMode, location.pathname, dispatch, notifyApiRequestFinished]);
 
+  const onSaveButtonClick = async () => {
+    setIsRequestSaving(true);
+
+    const record: Partial<RQAPI.ApiRecord> = {
+      id: apiRecordDeatils?.id,
+      type: RQAPI.RecordType.API,
+      data: entry,
+    };
+
+    const result = await upsertApiRecord(uid, record, teamId);
+
+    if (result.success) {
+      setApiRecordDetails(result.data as RQAPI.ApiRecord);
+    }
+
+    setIsRequestSaving(false);
+  };
```

Comment:
Show visual feedback (toasts) for success/failed saves.
Ignore this if this is going to be addressed in UI PR.